pathfinder/token_network.py

```diff
@@ -155,27 +155,28 @@ def update_balance(
         )
 
     def update_fee(
-        self,
-        channel_id: ChannelId,
-        new_fee: float,
-        signature: Address
+            self,
+            channel_id: ChannelId,
+            new_fee: bytes,
```

Comment:
I get that you probably don't want `new_fee` to be `float` if you need it as a signed message, but I guess using `str` here and `new_fee` being something like `'0.00058'` makes the most sense. It avoids the precision issue as well as the unique floating point value representation issue. I'm open to discussion on this one though, as there might be several solutions. I just don't think `bytes` is the prettiest.

Client-side it's also a lot easier to convert `float` into `str` than into `bytes`. But that's just my two cents.

Reply:
@devium Shouldn't we make this an `int` and the unit the token wad (apparently the wei for tokens)

Reply:
Bytes are necessary for proper hashing and signature verification. Bytes are constructed from string via str().encode, so no float problems here. Float is only used for internal computation and stored in channel_view.

Reply:
you should probably use `struct` if you are dealing with binary data.
https://github.com/raiden-network/raiden-libs/blob/master/raiden_libs/messages/balance_proof.py#L58

Reply:
regarding use of float, it's always better to use (fixed point) integer.

Reply:
@palango Depends on the fee model we decide to go with. Fractions totally make sense over absolute token values. Fixed point makes sense of course but for calculations it really doesn't matter. `float` precision is likely good enough for routing. So no issue with float vs fixed point but we need a deterministic representation for signature verification. `str` works in that regard (or the `bytes` representation thereof).